src/status_im2/contexts/chat/events.cljs

```diff
@@ -211,7 +214,7 @@
   [{db :db :as cofx} chat-id animation]
   (rf/merge cofx
             {:dispatch [(if animation :shell/navigate-to :navigate-to) :chat chat-id animation]}
-            (when-not (or (= (:view-id db) :community) (= (:view-id db) :community-overview))
+            (when-not (#{:community :community-overview :shell} (:view-id db))
```

Comment:
what is this list exactly? This is related to jump-to feature? Perhaps we can define in some constants file (or at least directly in this file) to give more context of it's use

Reply:
>This is related to jump-to feature?

This is not related to the jump-to feature. We by default pop to root when we open the chat screen, So that we dismiss any other screen. For example, if from a community channel, the user opens a profile and starts chatting with that user, the community will be dismissed. But we don't want this behavior when we are opening community chat, and if in a shell screen then there is not even anything to pop (but it will reset layout). Personally, it gives more context here instead of having it in a separate file.
```
(when-not (#{:community :community-overview :shell} (:view-id db))
(navigation/pop-to-root :shell-stack))
``` 
While opening chat, pop-to-root, if not navigating from these screens.

Reply:
ok thanks for the explanation. Just curious is there a `type` (or anything like that) for a "community-chat" and if so can we base this logic off this chat type rather than view id?

Reply:
Yes, there is a type for community chat. But in this case, "from" where we are navigating is important not "to" where. If we are navigating from the shell, it doesn't matter if we are navigating to 1-1 or community chat, there is nothing to pop, we are on root view.

Reply:
ok sounds good so, thanks @Parveshdhull  👌